tests/integration/utils2.go

```diff
@@ -1311,73 +1204,71 @@ func executeRequest(
 // failures are recorded properly. It will only yield once, once
 // the subscription has terminated.
 func executeSubscriptionRequest(
-	ctx context.Context,
-	t *testing.T,
-	allActionsDone chan struct{},
-	db client.DB,
-	testCase TestCase,
+	s *state,
 	action SubscriptionRequest,
-) (chan func(), bool) {
+) {
 	subscriptionAssert := make(chan func())
 
-	result := db.ExecRequest(ctx, action.Request)
-	if AssertErrors(t, testCase.Description, result.GQL.Errors, action.ExpectedError) {
-		return nil, true
-	}
-
-	go func() {
-		data := []map[string]any{}
-		errs := []error{}
-
-		allActionsAreDone := false
-		expectedDataRecieved := len(action.Results) == 0
-		stream := result.Pub.Stream()
-		for {
-			select {
-			case s := <-stream:
-				sResult, _ := s.(client.GQLResult)
-				sData, _ := sResult.Data.([]map[string]any)
-				errs = append(errs, sResult.Errors...)
-				data = append(data, sData...)
-
-				if len(data) >= len(action.Results) {
-					expectedDataRecieved = true
-				}
-
-			case <-allActionsDone:
-				allActionsAreDone = true
-			}
+	for _, node := range getNodes(action.NodeID, s.nodes) {
```

Comment:
question: It's not quite clear to me why we now have a for loop here. Could you give a quick explanation please?

Reply:
All of the other actions do this.  A test case may involve many nodes, is just this one and a couple of other stragglers hadn't implemented it yet.

Reply:
But it still worked and didn't affect outcome?

Reply:
The NodeID property was added along with the loop here.  Before it was essentially hardcoded to only ever act on the first node, regardless of the test configuration.

Reply:
These test were never meant to have more than one node but it could be useful in the future 👍